src/js/tech/html5.js

```diff
@@ -119,6 +121,125 @@ class Html5 extends Tech {
     super.dispose();
   }
 
+  /**
+   * A special function to trigger source set in a way that
+   * will allow player to re-trigger if player/html5 are not ready
+   * yet.
+   *
+   * @fires Tech#sourceset
+   */
+  triggerSourceSet_() {
+    if (!this.isReady_) {
+      // on initial ready we have to trigger source set
+      // 1ms after ready so that player can watch for it.
+      this.one('ready', () => this.setTimeout(this.triggerSourceSet_, 1));
+    }
+
+    /**
+     * Fired when the source is set on the tech causing the media element
+     * to reload.
+     *
+     * @event Tech#sourceset
+     * @type {EventTarget~Event}
+     */
+    this.trigger('sourceset');
+  }
+
+  /**
+   * Modify the media element so that we can detect when
+   * the source is changed. Fires `sourceset` just after the source has changed
+   */
+  watchForSourceSet_() {
+    if (!this.featuresSourceset) {
+      return;
+    }
+
+    const el = this.el();
+
+    // we need to fire sourceset when the player is ready
+    // if we find that the media element had a src when it was
+    // given to us and that tech element is not in a stalled state
+    if (el.src || el.currentSrc && this.el().initNetworkState_ !== 3) {
+      this.triggerSourceSet_();
+    }
+
+    const proto = window.HTMLMediaElement.prototype;
+    let srcDescriptor = {};
+
+    // preserve getters/setters already on `el.src` if they exist
+    if (Object.getOwnPropertyDescriptor(el, 'src')) {
+      srcDescriptor = Object.getOwnPropertyDescriptor(el, 'src');
+    } else if (Object.getOwnPropertyDescriptor(proto, 'src')) {
+      srcDescriptor = mergeOptions(srcDescriptor, Object.getOwnPropertyDescriptor(proto, 'src'));
+    }
+
+    if (!srcDescriptor.get) {
+      srcDescriptor.get = function() {
+        return proto.getAttribute.call(this, 'src');
+      };
+    }
+
+    if (!srcDescriptor.set) {
+      srcDescriptor.set = function(v) {
+        return proto.setAttribute.call(this, 'src', v);
+      };
+    }
+
+    if (typeof srcDescriptor.enumerable === 'undefined') {
+      srcDescriptor.enumerable = true;
+    }
+
+    Object.defineProperty(el, 'src', {
+      get: srcDescriptor.get.bind(el),
+      set: (v) => {
+        const retval = srcDescriptor.set.call(el, v);
+
+        this.triggerSourceSet_();
+
+        return retval;
+      },
+      configurable: true,
+      enumerable: srcDescriptor.enumerable
+    });
+
+    const oldSetAttribute = el.setAttribute;
+
+    el.setAttribute = (n, v) => {
+      const retval = oldSetAttribute.call(el, n, v);
+
+      if (n === 'src') {
+        this.triggerSourceSet_();
+      }
+
+      return retval;
+    };
+
+    const oldLoad = el.load;
+
+    el.load = () => {
+      const retval = oldLoad.call(el);
+
+      this.triggerSourceSet_();
```

Comment:
so, because we're now calling this event `sourceset` and the [html5 spec](https://html.spec.whatwg.org/multipage/media.html#dom-media-load) says that everything needs to be aborted and the resource selection algorithm is re-run, we want to trigger a `sourceset` here?
This is definitely something we'll want to document.

Reply:
Exactly. I think we could use a comment here and also a separate doc pr